src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -410,6 +467,93 @@ public function doSave($data, $form)
         return $this->redirectAfterSave($isNewRecord);
     }
 
+    /**
+     * Goes to the previous record
+     * @param  array $data The form data
+     * @param  Form $form The Form object
+     * @return HTTPResponse
+     */
+    public function doPrevious($data, $form)
+    {
+        Controller::curr()->getResponse()->addHeader("X-Pjax", "Content");
+        $link = $this->getEditLink($this->getPreviousRecordID());
+        return Controller::curr()->redirect($link);
+    }
+
+    /**
+     * Goes to the next record
+     * @param  array $data The form data
+     * @param  Form $form The Form object
+     * @return HTTPResponse
+     */
+    public function doNext($data, $form)
+    {
+        Controller::curr()->getResponse()->addHeader("X-Pjax", "Content");
+        $link = $this->getEditLink($this->getNextRecordID());
+        return Controller::curr()->redirect($link);
+    }
+
+    /**
+     * Creates a new record. If you're already creating a new record,
+     * this forces the URL to change.
+     *
+     * @param  array $data The form data
+     * @param  Form $form The Form object
+     * @return HTTPResponse
+     */
+    public function doNew($data, $form)
+    {
+        return Controller::curr()->redirect(Controller::join_links($this->gridField->Link("item"), "new"));
+    }
+
+    /**
+     * Gets the edit link for a record
+     *
+     * @param  int $id The ID of the record in the GridField
+     * @return string
+     */
+    public function getEditLink($id)
+    {
+        return Controller::join_links(
+            $this->gridField->Link(),
+            "item",
+            $id,
+            '?gridState=' . urlencode($this->gridField->getState(false)->Value())
+        );
+    }
+
+    /**
+     * Gets the ID of the previous record in the list.
+     *
+     * @return int
+     */
+    public function getPreviousRecordID()
+    {
+        $gridField = $this->gridField;
+        $gridStateStr = $this->getRequest()->postVar('gridState');
+        $gridField->getState(false)->setValue($gridStateStr);
+
+        $map = $gridField->getManipulatedList()->limit(PHP_INT_MAX, 0)->column('ID');
+        $offset = array_search($this->record->ID, $map);
+        return isset($map[$offset-1]) ? $map[$offset-1] : false;
+    }
+
+    /**
+     * Gets the ID of the next record in the list.
+     *
+     * @return int
+     */
+    public function getNextRecordID()
+    {
+        $gridField = $this->gridField;
+        $gridStateStr = $this->getRequest()->postVar('gridState');
+        $gridField->getState(false)->setValue($gridStateStr);
+
+        $map = $gridField->getManipulatedList()->limit(PHP_INT_MAX, 0)->column('ID');
```

Comment:
Goes without saying, but same change applies here. I think you'll want to encapsulate it into a method, like `$this->getCurrentPageWithAdjacentRecords()`